### ISSUE TITLE
Feature request: open in sidebar

### ISSUE THREAD

**Issue by marczellm** [olsh/Feedly-Notifier#129]
As a user, I would like to see the list of Feedly articles in a Firefox sidebar and open the full articles (on their website) in new tabs from there. (My current flow is: split the screen, snap FF to the right and a separate feedreader app to the left.)

Would this be possible? I can work on the implementation too and then pull request.

**Comment by olsh** [comment]
Hi @marczellm,

Sounds like a great idea, I would very appreciate if you submit a PR. 
Are you talking about [the API](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/user_interface/Sidebars)?

**Comment by marczellm** [comment]
Yes. I don’t know though if it would cause compatibility issues with other browsers or if those would just ignore it

**Comment by olsh** [comment]
It seems like the API is not supported in Chrome, so we should use preprocessing [like this](https://github.com/olsh/Feedly-Notifier/blob/master/src/manifest.json#L51)

**Comment by marczellm** [comment]
I started a fork: https://github.com/marczellm/Feedly-Notifier/tree/feature/firefox-sidebar
Most things seems to work smoothly because popup UI and sidebar UI is equivalent in how it works in Firefox.

The main issue is that while popup width is set by the extension, sidebar width is user resizable. Opening the summary of an article makes the UI wider than the max possible sidebar width. For sidebar, the UI should be made width-responsive. Unfortunately I have very little experience with CSS.

**Comment by olsh** [comment]
Thanks, I can try to help with this when I have some time, but probably this won't happen soon.

**Comment by marczellm** [comment]
Quoting https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/user_interface/Popups:

>  If the user moves the extension's button to the menu or it appears in the toolbar overflow, then the popup appears inside the menu's panel and is given a fixed width.

This means it looks like this:

![image](https://user-images.githubusercontent.com/6771275/44148167-6784ecd2-a097-11e8-93b4-782d11f68f80.png)

There is no way to scroll to the right hand side buttons.

Don't you think that it would be better if the entire UI would resize instead of having a fixed width? I tested and it would only require removing explicitly set widths in `popup.js`. Would that cause problems in other browsers?

**Comment by olsh** [comment]
This is implemented.

Firefox gets a real sidebar through `sidebar_action` (#386, made MV3-compatible in #390), and Chromium gets the equivalent through `side_panel` (#356). Both point at `popup.html?panel=1`, which is how the popup code tells a panel document from the toolbar popup, and there is a "Show in sidebar" option on the options page.

@marczellm's objection from 2018 — that the fixed popup width would break in a user-resizable sidebar — is handled: `setPopupWidth` skips setting a width when the document is the panel, so the UI follows the frame as you drag it.

The Chromium side panel shipped in v3.2.0. The Firefox sidebar is on master and lands in the next release.

**Comment by marczellm** [comment]
(The Firefox sidebar was originally implemented in #130, only not linked into this issue.)
